app/src/main/java/com/cornellappdev/android/eatery/components/WaitTimesComponent.java

```diff
@@ -67,6 +75,43 @@ public void onClick(View v) {
         });
         mWaitTimesChart = view.findViewById(R.id.wait_time_chart);
         mWaitTimesChart.setNoDataText("");
+
+        mWaitTimesChart.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                if (mFingerJustTapped) {
+                    // Measure the yDiff directly after the tap, high y diff means motion was a
+                    // vertical swipe
+                    float yDiff = Math.abs(event.getY() - mLastY);
+                    float xDiff = Math.abs(event.getX() - mLastX);
+                    // If it was not a vertical swipe
+                    if (xDiff > yDiff) {
+                        scrollView.requestDisallowInterceptTouchEvent(true);
+                        mVerticalScrolling = false;
+                    } else {
+                        // This variable (when true) is used below in onValueSelected
+                        // to prevent interaction with the mWaitTimesChart. I couldn't find
+                        // a cleaner way of disabling interaction with the chart
```

Comment:
![image](https://user-images.githubusercontent.com/43295248/67641288-ae73ea00-f8d7-11e9-9029-c7234a436304.png)

Also there are these methods from MPAndroidChart that could possibly make this cleaner?

Reply:
Fair point - I did look into a lot of these built in methods and I remember none seemed to work out at the time. The current solution works and I don't think it's a bad solution, so I'm resolving this for now.